text/083-boost-stdlib-dev/text.md

```diff
@@ -0,0 +1,774 @@
+    MSets/MSets.v, MSets/MSetWeakList.v, MSets/MSetGenTree.v,
+    MSets/MSetAVL.v, MSets/MSetInterface.v, MSets/MSetFacts.v,
+    MSets/MSetProperties.v, MSets/MSetRBT.v, MSets/MSetPositive.v,
+    MSets/MSetToFiniteSet.v, MSets/MSetEqProperties.v</details>
+* coq-roots-compat
+  * The Coq standard library, compatibility files
+  * Some compatibility files with older versions. Namespace `Coq.Compat`.
+  * 5 files, 0.1 kloc
+  * <details><summary>contains:</summary> Compat/AdmitAxiom.v,
+    Compat/Coq817.v, Compat/Coq819.v, Compat/Coq818.v,
+    Compat/Coq820.v</details>
+* coq-roots-derive
+  * The Coq standard library, derive
+  * Namespace `Coq.derive`
+  * 1 file, 0.0 kloc
+  * <details><summary>contains:</summary>
+    theories/derive/Derive.v</details>
+* coq-roots-rtauto
+  * The Coq standard library, rtauto
+  * Namespace `Coq.rtauto`
+  * 2 files, 0.8 kloc
+  * <details><summary>contains:</summary> rtauto/Bintree.v,
+    rtauto/Rtauto.v</details>
+* coq-roots-wellfounded
+  * The Coq standard library, wellfounded
+  * Namespace `Coq.wellfounded`
+  * 9 files, 0.9 kloc
+  * <details><summary>contains:</summary> Wellfounded/Inclusion.v,
+    Wellfounded/Wellfounded.v, Wellfounded/Union.v,
+    Wellfounded/Transitive_Closure.v, Wellfounded/Well_Ordering.v,
+    Wellfounded/Inverse_Image.v, Wellfounded/Disjoint_Union.v,
+    Wellfounded/Lexicographic_Product.v,
+    Wellfounded/Lexicographic_Exponentiation.v</details>
+* coq-roots
+  * The Coq standard library
+  * A metapackage for all coq-roots-* packages constituting
+    the historical Coq standard library, that used to be shipped with Coq.
+  * 508 files, 183.4 kloc
+
+theories/ssr and theories/ssrmatching should go back into mathcomp
+(either in coq-mathcomp-ssreflect or in a new coq-mathcomp-ssr
+containing only that).
+
+### Documentation
+
+Need to document the above split in packages (could help structure the doc).
+
+## Maintainers
+
+We should do an open call for maintainers. Current maintainers (stdlib
+team from Coq on github) will become roots maintainer, they can of
+course opt out at any time.
+
+# Drawbacks
+
+The stdlib will no longer come bundled with Coq. It will however be
+just as easy to install via the usual package managers. More notably
+it will remain an integral part of the Coq Platform.
+
+# Alternatives
+
+Other split in multiple repositories would be possible, could still
+happen in the future, maybe not needed now. The extreme opposite would
+be to lean toward a huge monorepo with everything from the platform,
+probably not very practical, and wouldn't bring much since the
+platform already ensures uniform releases bundling everything.
+To our knowledge the monorepo approach has exactly one advantage:
+it is easier to backport changes from one package to an upstream package.
+It also has one major drawback: it makes it harder to have the following dependencies:
+`coq-roots-A <- coq-extralib-B <- coq-roots-C`.
```

Comment:
I think an important alternative to consider here is performing the proposed logical-level refactoring in the current repository, without the reimagining the distribution model or conceptual role of the standard library. It would still achieve the concrete benefits for users who only want to import a part of it, retain the refactoring that has already been performed for this CEP, but minimize impact on users from further refactoring between the hard-won component boundaries. If individual components do diverge or want to add external dependencies as foreseen here, they could be spun out to their own packages as needed.

I also have opinions about the motivating vision for the future of stdlib, but to keep this discussion here focused on the component structure, I intend to post a whole another CEP for that in time for the scheduled Coq-call slot.

Reply:
Yes, please fill the "Alternatives" section of the CEP with your proposal and what you perceive are its pros and cons.

> I also have opinions about the motivating vision for the future of stdlib, but to keep this discussion here focused on the component structure, I intend to post a whole another CEP for that in time for the scheduled Coq-call slot.

Thanks, that will definitely be the next discussion to have once the current one gets resolved, and a CEP is probably a good way to drive the discussion.

Reply:
And by the way, no hurry to create the next CEP before discussing the current one in Coq Call. Those are two different discussions.

Reply:
> How can I add something to the alternatives section?

I'd expect you, as a maintainer, to be able to simply push to the branch of the CEP. If not, please ping me so that I give you proper rights.

> If these are different discussions, would you be willing to remove the nontechnical decisions about the future of stdlib from this CEP? Otherwise the proposed discussion schedule feels a little bit backwards.

Sure, those belong to a further CEP. The goal of the current one is only to unstuck the development of the current stdlib and "Enabling the development of an evolved or different official "stdlib"". Which one of the latter options we want to choose is outside the scope of the current CEP, we should only keep all doors open.

Reply:
GitHub thinks that I do not have permissions to edit coq/ceps or your branch.